test/mpi/rma/rma_contig.c

```diff
@@ -33,20 +33,17 @@ static void run_test(int lock_mode, int lock_assert)
 
         printf("Synchronization mode: ");
 
-        switch (lock_mode) {
-            case MPI_LOCK_EXCLUSIVE:
-                printf("Exclusive lock");
-                break;
-            case MPI_LOCK_SHARED:
-                printf("Shared lock");
-                break;
-            default:
-                printf("Unknown lock");
-                break;
+        if (lock_mode == MPI_LOCK_EXCLUSIVE) {
+            printf("Exclusive lock");
+        } else if (lock_mode == MPI_LOCK_SHARED) {
+            printf("Shared lock");
+        } else {
+            printf("Unknown lock");
```

Comment:
The C type for these constants must be `const int (or unnamed enum)` (pg. 859 MPI 4.0). As such, they should be perfectly fine to use with `switch...case`.

Reply:
IIRC, `const int` is a compile time constant in C++, but not in C :disappointed: .

Reply:
They are macro constants in MPICH - https://github.com/pmodels/mpich/blob/f51b6595223274eed72b3c6fcd5d8b49acd4b439/src/include/mpi.h.in#L396-L397

I wonder what was the issue 😕

Reply:
> I wonder what was the issue

Yes, although there is nothing wrong with the way MPICH does it. Users have to be careful. As the MPI standard says that "MPI constants" can be `const int`, portable MPI codes should not use them in `switch` statements. If the MPICH testsuite wants to be implementation-agnostic, then it should not use `switch statements with predefined/named handles and constants.

Reply:
Ah, thanks! I forgot to step outside MPICH.

Reply:
Yes, my goal is to use the MPICH test suite for other implementations.